Feat/edit project basics

therealemino opened the pull request:
# Edit Project Basics

## Description
This PR implements the ability to edit project details as outlined in issue #188. All acceptance criteria have been met.

## Changes Implemented
- Added a form to update project details.
- Editable text fields for Company Name, Tagline and social links.
- Integrated file upload (image and video) functionality.
- Updated `Input` component `components/base/input` to factor in prefixes (InputAdornment). This was to enable appending the social media icons in the input fields.
- Social media input fields.
- "Save Changes" button functionality.
- Ensured a11ycompliance.

## A few things to note:
### File Upload
- Implemented a file upload service using **Supabase**.

**I abstracted the file upload logic into:**
- A custom hook: `hooks/useFileUpload.ts` (`useUploadFile`). This validates the file, handles the state and returns state values (uploading, error etc) when uploading the file and error messages too.
- A helper function: `lib/utils/uploadFileToBucket.ts` (`uploadToBucket`). The actual upload logic that uploads the file to supabase.
- Since the Supabase buckets are not yet available, I'm temporarily returning a dummy image URL as the uploaded file URL until the buckets are set up.

### Commit Linting Issue
- Renamed `commitlint.config.js` to `commitlint.config.cjs`.
- Reason: Committing changes was throwing an error due to **ES module scope**.
- The project’s `package.json` contains `"type": "module"`, making `.js` files ES modules by default.
- Changing the extension to `.cjs` allows `commitlint` to be treated as a CommonJS module, resolving the issue.

### Testing & a11y
- Tested on Chrome and Edge
- Conducted accessibility testing on the form.
- Verified compliance with a11y best practices.

## Demo
A [video](https://vimeo.com/1059372253/62502785e1?share=copy) demonstrating the working flow of the edit form is attached for reference.

## Notes
- Please note that the file upload logic will need to be updated once Supabase buckets are configured.

## Related Issue
Closes #188 


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Launched a new project editing interface with robust form validation and streamlined media upload support.
  - Introduced enhanced file upload functionality for smoother file management.

- **Enhancements**
  - Upgraded form elements (inputs, textareas, and selects) with improved transition effects and interactive feedback.
  - Refined tag management and FAQ interactions for a more intuitive user experience.
  - Reintegrated key document upload components to enhance processes in identity verification workflows.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Bran18 reviewed (changes requested):
@therealemino I noticed the build is not working, can you please address it? 

<img width="1253" alt="Screenshot 2025-02-22 at 8 19 51 PM" src="https://github.com/user-attachments/assets/2a908cd4-79f4-4f8f-88c1-7a7bbc8acc4d" />


CC. @AndlerRL  can you please take a look at this PR I noticed some changes in files that I don't think we need to change

AndlerRL commented on apps/web/app/edit-project/page.tsx:
We can improve this. When importing a major page component such this one, we can only return the component. It is not required to have a wrapping `div` to place it, we can just:

```tsx
export default function EditProjectTestPage() {
        return <EditProjectForm />
```

Using `React.Fragment`/`<></>` tags might work but only if there is a logic like a render condition for the components.

AndlerRL commented on apps/web/components/sections/project/edit-project.tsx:
```diff
@@ -0,0 +1,707 @@
+	type EditProjectFormData,
+	MAX_LENGTHS,
+	editProjectFormSchema,
+} from '../../../lib/validators/project'
```
Keep in mind to be consistent in the exports to use `~/` when they are nested. Extensions like `auto-imports` for VSCode IDE might be helpful to set that easier.

AndlerRL commented on apps/web/lib/validators/project.ts:
```diff
@@ -2,6 +2,10 @@ import * as z from 'zod'
+	description: 500,
+}
 
 export const projectFormSchema = z.object({
```
Can we add tags and categories for the project schema? They are required to add in each project form.

AndlerRL reviewed (changes requested):
Excellent touch with the bucket upload! Good validation with zod and project structure in general. I noticed some files are `camelCase` and they should be `kebab-case` such as `useFileUpload` - to -> `use-file-upload.ts`. Do the same for the rest for camelCase files. I noticed that we are missing 2 fields in the form: category and tags. There is a [tags component ](https://kindfi-azhve2l7b-kindfi.vercel.app/project/create) already, you may place it below the description field. The categories should be a selector with a few options to pick (can be the same as the landing).

You are open to modify the tags UI to match them with your work if possible 🙌 like using [badges](https://ui.shadcn.com/docs/components/badge) components (we may have it already) and change the colour based on their tag values.

Additionally, I added some observations, please take them a look.

therealemino commented:
Hi @AndlerRL,
I've added the category and tag fields. Also reworked on the tag component (style and functionality (removed nested forms)).
All other requested changed have been implemented too.

![Screenshot 2025-02-24 021948](https://github.com/user-attachments/assets/7163ff15-ac8a-42e3-b0d3-8493c84d480f)
![Screenshot 2025-02-24 022039](https://github.com/user-attachments/assets/96645233-b78d-44a1-9e28-bc37394a79b8)
![Screenshot 2025-02-24 022021](https://github.com/user-attachments/assets/c54c1b3d-9818-4e12-bf3c-1869bdac5426)

Bran18 commented on bun.lockb:
@therealemino try run bun i again to fix this lock please

therealemino replied:
Alright. Will do this in a few hrs 🙏 (I'll update you once done).

Bran18 reviewed (changes requested):
It looks like you will need to run the `bun install` again to fix the lock file that is inside this PR @therealemino

AndlerRL reviewed (dismissed):
LGTM! Thanks for dealing with the request and retails :+1: only is missing to solve @Bran18 observation (should fix the build too) and we will be good to go!

therealemino commented:
Hi @Bran18,

Fixed the build error. 

Also I deleted my node_modules and tried bun install (3 times) and there were no changes in the bun.lockb.

So I just deleted the file and reinstalled. It generated a bun.lock file.
I think in general the text based lockfile (bun.lock) is more recommended. [See here](https://github.com/oven-sh/bun/issues/5486)

AndlerRL commented:
> Hi @Bran18,
> 
> Fixed the build error.
> 
> Also I deleted my node_modules and tried bun install (3 times) and there were no changes in the bun.lockb.
> 
> So I just deleted the file and reinstalled. It generated a bun.lock file. I think in general the text based lockfile (bun.lock) is more recommended. [See here](https://github.com/oven-sh/bun/issues/5486)
> 
> CC: @AndlerRL

Thanks for the knowledge, @therealemino! I read the issue thread and it makes sense to have no sense with the `bun.lockb` file! It should be for each machine and having only the `.lock` file in the repo. I will be making the according changes to the .gitignore and providing indications for the rest of the team 💪 I will be solving the conflicts for this one, the build went through 🙂

AndlerRL reviewed (approved):
LGTM! Thanks for the bun.lock suggestion, now the .gitignore is updated with the lockb file within it 👍 the changes looks very good and clean. Merging!

> Since I did a merge from develop, my biome.js detected some issues coming from develop, reason why we see more changes than the expected here now that it is sync.
